packages/kolibri/components/pages/AppBarPage/internal/AppBar.vue

```diff
@@ -207,24 +212,43 @@
     data() {
       return {
         pointsDisplayed: false,
+        appBarWidth: 0,
+        overflowCount: 0,
       };
     },
     computed: {
       // temp hack for the VF plugin
       usernameForDisplay() {
         return !hashedValuePattern.test(this.username) ? this.username : this.fullName;
       },
+      showTopNavBar() {
+        return this.overflowCount === 0;
+      },
+      truncatedTitle() {
+        if (!this.title) return '';
+        // Dynamically truncate title based on remaining space in AppBar
+        const offset = this.$refs.appBarActions?.clientWidth + 100;
+        const averageCharWidth = 10;
```

Comment:
I am interpreting averageCharWidth to be the average number of pixels per character in a string - is that right?

Reply:
Overall, I think this title truncation is 👍 - thanks for taking some additional time to tweak the approach

Reply:
Yes, that's exactly it. Dividing by it returns how many letters can be displayed before the title needs to be shortened.

I found that an average of 10 allows the title to be truncated while maximizing it’s visibility. With higher numbers, it truncates the title too soon, and with numbers lower than 10, the truncation is delayed and pushes the profile icon and username out of view.

Reply:
and the `100`, is it an arbitrary additional offset to cater for something in the app bar actions?

Reply:
The `appBarActions.clientWidth` alone did not add enough space between the title and the actions, but it does account for the width of the appbar actions. If the `100` is removed, or set to a lower number, everything on the right side of the AppBar (like the profile icon and username) will collide with the title sooner than it should.